src/pageObjects/WelcomePage.js

```diff
@@ -0,0 +1,44 @@
+
+
+
+export default class WelcomePage {
+    constructor(page){
+        this.page = page
+        this.url = ' / '
+        this.signInButton = page.locator('.hero-descriptor button');
+
+        // Registration form locators
+        this.headerRegistrationForm = page.locator('h4.modal-title');
+
+        this.nameField = page.locator('#signupName');
+        this.lastName =page.locator('#signupLastName');
+        this.password = page.locator('#signupPassword');
+        this.repeatPassword = page.locator('#signupRepeatPassword');
+        this.email = page.locator('#signupEmail');
+        this.registrButton = page.locator('.modal-footer button');
+
+        this.validationErrorLocator = page.locator('.invalid-feedback p');
```

Comment:
I would recommend you to create separated component SigUpPopup with these fields